packages/mermaid/src/rendering-util/handle-markdown-text.js

```diff
@@ -65,30 +59,27 @@ export function markdownToLines(markdown) {
 }
 
 /**
- *
- * @param markdown
+ * @param {string} markdown markdown to convert to HTML
+ * @returns {string} HTML
  */
 export function markdownToHTML(markdown) {
-  const mdParse = SimpleMarkdown.defaultBlockParse;
-  const syntaxTree = mdParse(markdown);
+  const { children } = fromMarkdown(markdown);
 
   /**
-   *
-   * @param node
+   * @param {import('mdast').Content} node
    */
   function output(node) {
     if (node.type === 'text') {
-      return node.content.replace(/\n/g, '<br/>');
+      return node.value.replace(/\n/g, '<br/>');
     } else if (node.type === 'strong') {
-      return `<strong>${node.content.map(output).join('')}</strong>`;
-    } else if (node.type === 'em') {
-      return `<em>${node.content.map(output).join('')}</em>`;
+      return `<strong>${node.children.map(output).join('')}</strong>`;
+    } else if (node.type === 'emphasis') {
+      return `<em>${node.children.map(output).join('')}</em>`;
     } else if (node.type === 'paragraph') {
-      return `<p>${node.content.map(output).join('')}</p>`;
-    } else {
-      return '';
+      return `<p>${node.children.map(output).join('')}</p>`;
     }
+    return `Unsupported markdown: ${node.type}`;
```

Comment:
Only logic change is here. 
This will make the user aware that they are using unsupported markdown. 
Can remove once micromark is added to support complete html.

Reply:
Sounds good!